File: Aug14/Troy/Q11.java
class Solution {
    public int maxArea(int[] height) {
        int len = height.length;
        int result = 0;
        int start = 0;
        int end = len - 1;
        while (start < end) {
            int temp = (end - start) * Math.min(height[start], height[end]);
            if (temp > result) result = temp;
            if (height[start] < height[end]) {
                start++;
            } else {
                end--;
            }
        }
        return result;
    }
}

File: Aug14/Carl/Q11.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        if len(height) == 0:
            return 0
        left = 0
        right = len(height) - 1 
        maxArea = 0 
        while left != right:
            maxArea = max(maxArea, (right-left)*min(height[right], height[left]))
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        return maxArea

S = Solution()
print(S.maxArea([1,2]))